core/system/src/unvic.c

```diff
@@ -443,10 +443,16 @@ void __unvic_svc_cx_out(uint32_t *svc_sp, uint32_t *msp)
 
 void unvic_init(void)
 {
+    uint8_t prio_bits;
+    uint8_t volatile *prio;
+
     /* Detect the number of implemented priority bits. */
     __disable_irq();
-    *((volatile uint8_t *) &(NVIC->IP[0])) = 0xFFU;
-    g_nvic_prio_bits = (uint8_t) __builtin_popcount(*((volatile uint8_t *) &(NVIC->IP[0])));
+    prio = (uint8_t volatile *) &(NVIC->IP[0]);
+    prio_bits = *prio;
+    *prio = 0xFFU;
```

Comment:
@AlessandroA i would add a comment here on why we're doing the next two lines ("the architecture specifies that unused bits read back as zeroes" or similar)